iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -50,9 +51,14 @@ public async Task InitLoopAsync(CancellationToken cancellationToken)
                     cancellationToken)
                 .ConfigureAwait(false);
 
+            // InitLoopAsync is not re-entrant, so we can initialize a new cancellation token source for each
+            // instance of AmqpAuthenticationRefresher initialized.
+            // This cancellation token source is disposed when the authentication refresher is disposed.
+            _refresherCancellationTokenSource = new CancellationTokenSource();
```

Comment:
the original cancellation token passed into this API depends on the cancellation token of the upstream call, i.e. the cancellation token passed into OpenAsync() etc. I'd like to have that decoupled since the token refresher should ideally run on a separate thread all the way until it is canceled.